src/overlay073/ov73_021D0D80.c

```diff
@@ -1091,1155 +1549,1459 @@ static const u8 Unk_ov72_021D39D4[] = {
-            param0->unk_7C = 6;
-            param0->unk_80 = 10;
-            param0->unk_78 = 2;
+            manager->fadeBgLayerCurAlpha = 6;
+            manager->fadeBgLayerCurAlphaInv = 10;
+            manager->fadeBgLayerState = FBL_STATE_FADE_OUT;
 
-            if (param0->unk_84 == 0) {
-                param0->unk_0C = 68;
+            if (manager->playerGender == GENDER_MALE) {
+                manager->state = RI_STATE_GENDR_FADE_OUT_AVATAR_FEMALE;
             } else {
-                param0->unk_0C = 70;
+                manager->state = RI_STATE_GENDR_FADE_OUT_AVATAR_MALE;
             }
             break;
         }
 
         if ((gSystem.pressedKeys & PAD_KEY_LEFT) == PAD_KEY_LEFT || (gSystem.pressedKeys & PAD_KEY_RIGHT) == PAD_KEY_RIGHT) {
-            if (param0->unk_84 == 0) {
-                param0->unk_84 = 1;
+            if (manager->playerGender == GENDER_MALE) {
+                manager->playerGender = GENDER_FEMALE;
             } else {
-                param0->unk_84 = 0;
+                manager->playerGender = GENDER_MALE;
             }
 
             Sound_PlayEffect(SEQ_SE_CONFIRM);
         }
 
-        ov73_021D1DE8(param0);
+        RowanIntro_AnimateAvatarRun(manager);
         break;
-    case 68:
-        if (ov73_021D1334(param0, 2, 1) == 1) {
-            param0->unk_0C = 69;
+    case RI_STATE_GENDR_FADE_OUT_AVATAR_FEMALE:
+        if (RowanIntro_FadeBgLayer(manager, BG_LAYER_MAIN_2, FADE_OUT) == TRUE) {
+            manager->state = RI_STATE_GENDR_CENTRE_AVATAR_MALE;
         }
         break;
-    case 69:
-        if (ov73_021D1D44(param0, 1, 0) == 1) {
-            ov73_021D1EFC(param0);
-            param0->unk_0C = 72;
+    case RI_STATE_GENDR_CENTRE_AVATAR_MALE:
+        if (RowanIntro_MoveBgLayer(manager, BG_LAYER_MAIN_1, MBL_CASE_CENTRE_AVATAR) == TRUE) {
+            RowanIntro_ResetBlending(manager);
+            manager->state = RI_STATE_GENDR_CONFIRM_DIALOGUE;
         }
         break;
-    case 70:
-        if (ov73_021D1334(param0, 1, 1) == 1) {
-            param0->unk_0C = 71;
+    case RI_STATE_GENDR_FADE_OUT_AVATAR_MALE:
+        if (RowanIntro_FadeBgLayer(manager, BG_LAYER_MAIN_1, FADE_OUT) == TRUE) {
+            manager->state = RI_STATE_GENDR_CENTRE_AVATAR_FEMALE;
         }
         break;
-    case 71:
-        if (ov73_021D1D44(param0, 2, 0) == 1) {
-            ov73_021D1EFC(param0);
-            param0->unk_0C = 72;
+    case RI_STATE_GENDR_CENTRE_AVATAR_FEMALE:
+        if (RowanIntro_MoveBgLayer(manager, BG_LAYER_MAIN_2, MBL_CASE_CENTRE_AVATAR) == TRUE) {
+            RowanIntro_ResetBlending(manager);
+            manager->state = RI_STATE_GENDR_CONFIRM_DIALOGUE;
         }
         break;
-    case 72: {
-        u32 v4;
+    case RI_STATE_GENDR_CONFIRM_DIALOGUE: {
+        u32 genderConfirmTextID;
 
-        if (param0->unk_84 == 0) {
-            v4 = 22;
+        if (manager->playerGender == GENDER_MALE) {
+            genderConfirmTextID = 22;
```

Comment:
```suggestion
            genderConfirmTextID = pl_msg_00000389_00022;
```
add `#include res/text/bank/text_bank_rowan_intro.h` to the top of the file. This applies to other textIDs too.
You have named this text bank, so you may want to give the text better names if you feel up to it.

Reply:
I have changed the other textIDs, though I may have missed a few.